src/world/World.h

```diff
@@ -148,6 +148,10 @@ class World : public MapBase
     /// Create the FOW-objects, -streets, etc for a point and player
     void SaveFOWNode(const MapPoint pt, const unsigned player, unsigned curTime);
     unsigned GetNumSeas() const { return seas.size(); }
+    // returns if a map point is surrounded by a given TerrainType. Result is evaluated by the given predicate.
```

Comment:
* Return (use command like description, shorter and a good convention)

Better for this: `Return if all surrounding terrains match the given predicate`